### PR TITLE
[DB-13678] Assess Migration command guardrails check for pgss extension properly loaded or not in source db

### PR DESCRIPTION
**Pull request by sanyamsinghal**
- doing SELECT query on pg_stat_statements view and comparing the error message

### CODE REVIEW

File: yb-voyager/src/srcdb/postgres.go [hunk 1]
```diff
@@ -1220,6 +1206,14 @@ func (pg *PostgreSQL) checkPgStatStatementsSetup() (string, error) {
 		return "User doesn't have permissions to read pg_stat_statements view, unsupported query constructs won't be detected/reported", nil
 	}
 
+	// To access "shared_preload_libraries" must be superuser or a member of pg_read_all_settings
+	// so instead of getting current_settings(), executing SELECT query on pg_stat_statements view
+	// 3. check if its properly installed/loaded without any extra permissions
+	_, err = pg.db.Exec(queryCheckPgssLoaded)
+	if err != nil && strings.Contains(err.Error(), "pg_stat_statements must be loaded via shared_preload_libraries") {
```
Comment (priyanshi-yb) [inline yb-voyager/src/srcdb/postgres.go:1213]: nit: error msg constant `SHARED_PRELOAD_LIBRARY_ERROR`

File: yb-voyager/src/srcdb/postgres.go [hunk 2]
```diff
@@ -1166,7 +1166,7 @@ const (
 		JOIN pg_namespace n ON e.extnamespace = n.oid
 		WHERE e.extname = 'pg_stat_statements'`
 
-	querySharedPreloadLibraries = `SELECT current_setting('shared_preload_libraries')`
+	queryCheckPgssLoaded = `SELECT query from pg_stat_statements LIMIT 1`
```
Comment (priyanshi-yb) [inline yb-voyager/src/srcdb/postgres.go:1169]: schema of pgss table in case it's in non_public.

Comment (sanyamsinghal) [reply yb-voyager/src/srcdb/postgres.go:1169]: good catch.

**Review by priyanshi-yb — APPROVED** [review]
LGTM

**Pull request merged**